### PR TITLE
UserAuth Logic Improvments and UI enhancement

### PR DESCRIPTION
**Pull request by Hemant2335**
## Related Issue

Closes: #234 

## Description of Changes

- [x] My code adheres to the established style guidelines of this project.
- [x] I have conducted a self-review of my code.
- [x] I have included comments in areas that may be difficult to understand.
- [x] I have made corresponding updates to the project documentation.
- [x] My changes have not introduced any new warnings.

## Screenshots

|           Original           |           Updated           |
| 
Screenshot (113)
| 
Screenshot (112)
|

### CODE REVIEW

**Review by anmode — CHANGES_REQUESTED** [review]
If you are using tailwind.
1. Remove the login css which is not needed.
2. can you do like this when user click on register button all things remains same that pic too but registration form comes in place of login 
3. Pic is taking time in loading ( Pleae reduce the size)
4. Just a question how to differentiate between tailwind css or our manual css??

**Comment by Hemant2335** [comment]
1.) Task Completed (Please check)
2.) Task Completed (Please check)
3.) Task Completed (Please check)
4.) Yaa before you asked me this question I was very confused in using the custom tailwind css classes , so the basic difference could be the utility and other classes provided by the library can help us to differentiate between custom css and such libraries
5.) I think it is quite responsive (Please check)

Thank you

Attached video


https://github.com/anmode/grabtern-frontend/assets/85656999/4529eac5-1a2c-4558-a4b8-47816b9fb9a9

**Comment by anmode** [comment]
uff! This UI is not good. Your Previous UI was damn awesome. and it is still not optimized for mobile Take you time. this is nice issue would be level 3.

thanks for explaining!

**Comment by Hemant2335** [comment]
I have reverted the design back to previous one and also removed the image in the mobile view but still there always some space left for vertical scroll please check

ss
Screenshot (114)
Screenshot (117)
Screenshot (116)
Screenshot (115)

**Comment by anmode** [comment]
see if you can do like this. I just wanna ask why you have used logstate? and these context??
```
import { useRouter } from 'next/router';

export default function AuthPage() {
  const router = useRouter();
  
  const handleSignIn = () => {
    // Handle sign-in logic here
    router.push('/auth/#signin');
  };
  
  const handleSignUp = () => {
    // Handle sign-up logic here
    router.push('/auth/#signup');
  };
  
  return (
    <div>
      <h1>Authentication</h1>
      
      <form>
        <h2>Sign In</h2>
        {/* Sign-in form fields */}
        <button type="submit" onClick={handleSignIn}>Sign In</button>
      </form>
      
      <form>
        <h2>Sign Up</h2>
        {/* Sign-up form fields */}
        <button type="submit" onClick={handleSignUp}>Sign Up</button>
      </form>
    </div>
  );
}

```

**Comment by Hemant2335** [comment]
Yaa well I think for this to be true we need to define the handle submit function for both of them on the same component / page

-> And I don't want to disturb the current structure so I left them separated and when then ( I think ) if we use router.push() it will
lead us directly to that page not the component

-> So I prefer rather using the context and define a one state for both pages and render when true and false

-> And also by the functions (handleSignup()) this the same handle submit button for sign up right ?

-> But your way is quite the more effective and simpler one if you want me to implement this one I can 

Thank you

**Comment by anmode** [comment]
Ok! This PR would be very great.. Thanks btw.
I changed some things following up my previous comment.
what you have to do:
1. Please see the that logic is all set or not?
2. Please fix the UI ( remove scrolling - Hint: Fix the image size)
3. Please optimize it for mobile.

thanks

**Comment by anmode** [comment]
I have kept in mind your comments. Please check current folder structure nd mine previous to previous comments.
I did now its your turn. Add correct logic and optimize it.

**Comment by Hemant2335** [comment]
-> I have checked my logic and I think it is all set and at all don't have any bugs but if in future if some error arises I will do my best to solve it 

-> And I have removed the vertical scrolling 

-> Also now there is no picture in mobile view 

Please check video

https://github.com/anmode/grabtern-frontend/assets/85656999/7439c28c-6b73-4b35-b349-a5d07cdbbd74

Thank you

**Comment by anmode** [comment]
I see building is failing....(If that is because of /mentors no worries I'll fix that)
1. Please make routing like as `grabtern.com/userAuth/#login` and `grabtern.com/userAuth/#register` .
2. I see there is scroll bar in mobile view.. Just fix the height or you reduce the form height. Actually form is very short so we do not scrolling hope you understand.
Ok. I'll verify logic by checking pR locally. You can do these changes. please

**Comment by Hemant2335** [comment]
-> I have fixed the mobile view scrolling 
-> And could you please tell me where I have to change the routing or are you going to check it 

Thank you

Attached ss
Screenshot (119)
Screenshot (121)
Screenshot (120)

**Comment by anmode** [comment]
So in userAuth page... see if this code works.
```
function UserAuthPage() {
  const router = useRouter();

  useEffect(() => {
    // Code to handle hash change
    handleHashChange();
  }, []);

  const handleHashChange = () => {
    const currentHash = router.asPath.split('#')[1];

    if (currentHash === 'login') {
      // Show login component
      // Add your logic here
    } else if (currentHash === 'register') {
      // Show register component
      // Add your logic here
    } else {
      // Invalid hash, handle accordingly (e.g., show an error message)
      // Add your logic here
    }
  };

  return (
    <div>
      {/* Add your user authentication component */}
    </div>
  );
}

export default UserAuthPage;
```

```
return (
  <div>
    {currentHash === 'login' && <LoginComponent />}
    {currentHash === 'register' && <RegisterComponent />}
    {currentHash !== 'login' && currentHash !== 'register' && (
      <ErrorMessageComponent />
    )}
  </div>
);
```

File: public/assets/img/gallery/loginimg.jpg
Comment (anmode) [inline public/assets/img/gallery/loginimg.jpg:1]: this image should be reduced... 1.28mb

File: public/assets/img/gallery/20944201.jpg
Comment (anmode) [inline public/assets/img/gallery/20944201.jpg:1]: see if this too can be reduced

File: pages/userAuth.page.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  return (
+    <>
+      <Header navbarBackground={true} />
+      <main className="login-body d-flex flex-row justify-content-between">
+        {logpagestate ? (
+          <Login handleLogPageToggle={handleLogPageToggle} />
+        ) : (
+          <Register handleLogPageToggle={handleLogPageToggle} />
+        )}
+        <div className="tw-hidden md:tw-flex tw-h-[700px]">
```
Comment (anmode) [inline pages/userAuth.page.js:29]: You have used `700px` I think it is good to use relative other option like`vw`

**Review by anmode — CHANGES_REQUESTED** [review]
Looks cool!

**Comment by Hemant2335** [comment]
> So in userAuth page... see if this code works.
> 
> ```
> function UserAuth() {
>   const router = useRouter();
> 
>   useEffect(() => {
>     // Code to handle hash change
>     handleHashChange();
>   }, []);
> 
>   const handleHashChange = () => {
>     const currentHash = router.asPath.split('#')[1];
> 
>     if (currentHash === 'login') {
>       // Show login component
>       // Add your logic here
>     } else if (currentHash === 'register') {
>       // Show register component
>       // Add your logic here
>     } else {
>       // Invalid hash, handle accordingly (e.g., show an error message)
>       // Add your logic here
>     }
>   };
> 
>   return (
>     <div>
>       {/* Add your user authentication component */}
>     </div>
>   );
> }
> 
> export default UserAuth;
> ```
> 
> ```
> return (
>   <div>
>     {currentHash === 'login' && <LoginComponent />}
>     {currentHash === 'register' && <RegisterComponent />}
>     {currentHash !== 'login' && currentHash !== 'register' && (
>       <ErrorMessageComponent />
>     )}
>   </div>
> );
> ```

So Sir could you please tell me  in this code which function should I will be using on the components buttons (Login , Register)  to refresh the changes or to change the components.

**Comment by anmode** [comment]
no worries.. I will add logic of it. in this PR.( As of now i also have too see what work hahah)
but one thing please seek for this... You can relative sizing nd fix the register and logic as still scrolling is there. Do not use px. you vh ,vw or other relative...
Secondly login page is good but register for not.. as form has large padding and content is compressed like...i tried to point in the video. If any problem feel free to ask
https://github.com/anmode/grabtern-frontend/assets/88332977/42d28632-4cb0-475b-970a-929c94115096

**Review by sayyedarib — APPROVED** [review]
from my side logic is correct and everything is working properly

- use of div should be avoided
- instead of jpg image webp can be used

**Comment by anmode** [comment]
I have fixed the logic of page rendering. You check it out. Last couple of UI improvements, please. Then PR will be good to merge. Thanks

**Comment by Hemant2335** [comment]
I have
-> Corrected the padding in the registration page
-> Also have made some changes to prevent vertical scrolling in mobile view please check

Attached video


https://github.com/anmode/grabtern-frontend/assets/85656999/1adb80c2-6fed-4d27-bfc8-3e08872cc24c

**Comment by anmode** [comment]
> I have -> Corrected the padding in the registration page -> Also have made some changes to prevent vertical scrolling in mobile view please check
> 
> Attached video
> 
>  2023-06-05.07-33-05.mp4

cool! please commit you changes so I can do a final review. and also in mobile mode, i see there is so much bottom space to keep the form in the center. and referring the @sayyedarib comment change `jpeg` to `webp` please otherwise it is good to merge thanks

**Comment by Hemant2335** [comment]
I have converted the image into webp

Thank you

**Review by anmode — APPROVED** [review]
Cool! Thank you for making this together. Congrats @Hemant2335

**Comment by anmode** [comment]
You can share the pR in gssoc group or any other places too and your experience with us. It would be a great help to the project and we hope you other UI improvements from your side
@Hemant2335

**Comment by Hemant2335** [comment]
Sir Could you please add Gssoc label on it because  this pr hasn't been counted in the Leaderboard

**Pull request merged**